### ISSUE TITLE
ZAP Proxy not working in MAC

### ISSUE THREAD

**Issue by jainchirag21064** [zaproxy/zaproxy#6162]
**Describe the bug**
Trying to install ZAP on MAC

**To Reproduce**
1. Go to ZAP Site
2. Click on MAC OS intaller
3. Install it

**Screenshots**

![image](https://user-images.githubusercontent.com/13871439/92360235-cb2ea000-f0ec-11ea-963e-6003590e8cf4.png)

**Additional context**
I also tried downloading it from HomeBrew bit getting same error

**Comment by thc202** [comment]
This is a duplicate of #5204, leaving this open as the other is already locked, but I expect the same fate.

**Comment by jgadsden** [comment]
We see the same issue for Threat Dragon, https://github.com/OWASP/threat-dragon-desktop/issues/43
If there is a solution then could Threat Dragon project also use it?

**Comment by kingthorin** [comment]
There are installer’less options. As long as Java is available on the system the cross platform or weekly release should work (I believe).

**Comment by jgadsden** [comment]
Yes, agreed @kingthorin , but I think this is a growing problem for MacOS users, and possibly also for Windows users.
Could we look again at signing for ZAP, and maybe other OWASP projects? It is complaint made against Threat Dragon as well.
In my company we use both ZAP and TD, and we are a Microsoft-free company, so this will soon be a big problem to us

**Comment by psiinon** [comment]
I actually did get a Mac developer cert and tried to sign ZAP but I was not able to get it to work.
The problem appeared to be related to the fact that ZAP is a java app and uses a separate java based installer.  Java isnt well supported on Macs. Oh, and that I'm no Mac expert so it could have just been my inexperience ;)
The Homebrew cask may be slightly easier as we dont have the installer in the way.
We can try again sometime, but no promises that we'll be able to fix this :/

**Comment by jgadsden** [comment]
Hello @psiinon - I could look into this again if you like? The reason I am keen on this is that it is also a problem for Threat Dragon (OWASP incubator project). My thinking is that users are more likely to persevere with a flagship project like ZAP, but for TD they may just junk TD straight away without trying it.
How did you get the Mac developer cert? was this through OWASP organisation?

**Comment by psiinon** [comment]
@jgadsden that would be great! I just signed up for the Mac developer program - I think its free, or very cheep. You can then generate your own developer cert.

**Comment by psiinon** [comment]
Note that for the installer we use Install4j - that does complicate things :/ But we can explain how thats built and try things out if you think you might have got something working :)

**Comment by jgadsden** [comment]
OK, thanks - I will ask that this is the focus for version 1.4 of Threat Dragon Desktop. If all goes well it may be that we can use something similar for ZAP

**Comment by jgadsden** [comment]
Similar issue for Amass, using precompiled binaries:
https://github.com/OWASP/Amass/releases

**Comment by Pheidari57** [comment]
I applied the same solution suggested [here:](https://apple.stackexchange.com/questions/366542/install-spotify-cant-be-opened-because-apple-cannot-check-it-for-malicious-so) 

xattr -d com.apple.quarantine /Applications/OWASP\ ZAP.app/
and finally, it works on my mac.

**Comment by jgadsden** [comment]
Hello @jainchirag21064 - could you try the following on your MacOS system? It may solve your issue, it worked for me:
* download the .dmg (I used the latest ZAP_2.9.0.dmg)
* open it and drag 'OWASP ZAP' to the application directory as normal, wait for copying to finish
* run ZAP from launchpad or Finder -> Applications
* most likely get fatal pop-up _“OWASP ZAP” cannot be opened because the developer cannot be verified, macOS cannot verify that this app is free from malware._
* close this message popup
* from the MacOS 'Apple', top left of display, go to "System Preferences" > "Security & Privacy"
* at the bottom of the dialog, see message saying that "OWASP ZAP" was blocked. Next to it, click "Open anyway"
* the initial error message will pop up again, but this time have the option to click "Open" to run the installer despite the warning
* click 'Open' and ZAP now runs OK
* this only has to be done once, after this ZAP can be run as normal

**Comment by jgadsden** [comment]
@psiinon - The solution above could be put in the installation section of ZAP documentation? I looked into getting a key from the Mac developer program but it wanted to charge me $90 or so. It makes me vaguely nauseous handing money over to Apple, so the above seems a better solution?

**Comment by psiinon** [comment]
Yes, we could definitely have a FAQ for this (to cover Windows as well) and then link to that from the Downloads page...

**Comment by Big-Juan** [comment]
Thanks for posting this,  OWASP ZAP is now running on my iMac with Big Sur!!

**Comment by jgadsden** [comment]
Thanks @Big-Juan , it has reminded me to put this in a FAQ for the docs. Will try to get the PR done soon ... tech writing is not my strong point though :-)

**Comment by dkrysiak** [comment]
> And we do publish the checksums - see the very first line on https://www.zaproxy.org/download/
Oh, sorry, didn't notice that, great!

And I'm glad there's a Windows issue. Now it's linked :)

**Comment by phydroxide** [comment]
It speaks to the trustworthiness of the app. It is harder and harder to know if the site you pull it down from is legit. Bypassing this gate is scary especially for security folk.

**Comment by jgadsden** [comment]
Hello @psiinon 

How big a problem is this for the ZAP community? We had a similar hill to climb woth Threat Dragon.

In Threat Dragon land we now have signed Windows installer and notarised MacOS install image, @leoreading did a lot of work to get  this into the release pipeline. Should we get something like this into ZAP?

**Comment by psiinon** [comment]
@jgadsden Yes please!
We have _started_ to look at this and its definitely something we want to do, but any help, advice and guidance you can give to us would be very gratefully received :D

**Comment by dwdraju** [comment]
> xattr -d com.apple.quarantine /Applications/OWASP\ ZAP.app/

This worked!

**Comment by RalphRuiz96** [comment]
This is to help those that still do not get @dwdraju, basically go to the application directory, where the Owasp Zapp.app is, and then run the command that was provided by dwdraju. From there you navigate to the OWASP\ ZAP.sh file and then run ./OWASP\ ZAP.sh  and thats it! I have also managed to download it using a virtual machine. I know it is a stretch to go in this route, but if this does not work then a virtual machine option is something that people could look into. I was actually surprised that the command provided above worked for me lol.

**Comment by phydroxide** [comment]
the dwdraju solution does not address the trust issue that causes the problem in the first place. What the Zap developers are saying is essentially - 'we are unable to package this up in a trustworthy form, so we'd ask our customers to please ignore this and install it anyway'. Yes working around the security does work. It remains a tad sus.

**Comment by psiinon** [comment]
Apple dont make it easy to sign Java apps. We've done some playing around and think we can get it done but these things take time and none of us are experts on MacOS signing. We are also doing _lots_ of other things. Contributions gratefully received :)

**Comment by phydroxide** [comment]
I predict that buried in the response to "What makes it hard?" is a task that relates to actual security... given there are so many ways to interfere with the integrity of a Java application. You might consider discussing here what trouble you're running into. I predict with more detail there will be more willing to help you through it.

**Comment by fzipi** [comment]
You can achieve the same using `--no-quarantine`:
```
❯ brew install --cask owasp-zap --no-quarantine
==> Downloading https://github.com/zaproxy/zaproxy/releases/download/v2.11.1/ZAP_2.11.1.dmg
Already downloaded: /Users/fzipi/Library/Caches/Homebrew/downloads/6d37e09775a1132930c8f55926c7af7ddd3d4a8894581e0dcfd7047387d38405--ZAP_2.11.1.dmg
==> Installing Cask owasp-zap
Warning: macOS's Gatekeeper has been disabled for this Cask  <===========
==> Moving App 'OWASP ZAP.app' to '/Applications/OWASP ZAP.app'
🍺  owasp-zap was successfully installed!
```

`--no-quarantine` ~= `xattr -d com.apple.quarantine /Applications/OWASP\ ZAP.app/`

**Comment by phydroxide** [comment]
... yeah but....

**Comment by montao** [comment]
I got it going this way 

`sudo xattr -d com.apple.quarantine /Applications/OWASP\ ZAP.app `

I need to solve signing for macOS with my own app as well. If I get it to work I can come back here and propose some once-and-for-all so that we don't need to manually approve it for every installation.

**Comment by psiinon** [comment]
We definitely want to sign the Mac OS (and Windows) installers / apps, and have done some initial investigations.
But its non trivial, so if anyone could help with it then that would make it much more likely to actually happen ;)

**Comment by Bulky642** [comment]
adding another solution that worked for me, on MACos Monterey version 12.4:
Download the MACos installer from zapproxy.org --> move installer to Applications --> Open finder, right click on ZAP and choose open, you will get a warning click open anyway and you are off to the races. 
( double clicking on ZAP didn't work for me in Applications).

**Comment by patrickjane** [comment]
> You can achieve the same using `--no-quarantine`:
> 
> ```
> ❯ brew install --cask owasp-zap --no-quarantine
> ==> Downloading https://github.com/zaproxy/zaproxy/releases/download/v2.11.1/ZAP_2.11.1.dmg
> Already downloaded: /Users/fzipi/Library/Caches/Homebrew/downloads/6d37e09775a1132930c8f55926c7af7ddd3d4a8894581e0dcfd7047387d38405--ZAP_2.11.1.dmg
> ==> Installing Cask owasp-zap
> Warning: macOS's Gatekeeper has been disabled for this Cask  <===========
> ==> Moving App 'OWASP ZAP.app' to '/Applications/OWASP ZAP.app'
> 🍺  owasp-zap was successfully installed!
> ```
> 
> `--no-quarantine` ~= `xattr -d com.apple.quarantine /Applications/OWASP\ ZAP.app/`

This worked for me.

I wanted to note, since there was a recommendation to choose the "Open anyway" option, that on our devices, enterprise management software has been installed, and thus this option is not available.

While the `--no-quarantine` option worked, it would be better though, to have an official / signed approach available.

**Comment by qqpann** [comment]
Using a security app that increases security risk (bypassing security checks) sounds bad. It would be great if it could be officially signed.

**Comment by qqpann** [comment]
Maybe we can notify about this situation in the download page when someone tries to download for Mac, or add a 

> **Note**: section-ish

in the download page, and ask for help?
By bringing more attention to this issue, we hopefully have a better chance of reaching someone who can fix it.
(I'm not professional about signing, and an amature in Java)

**Comment by yra-wtag** [comment]
For Any future Contributor with Apple Developer Account from ChatGPT

Signing a Java application DMG package for Apple involves a few steps to ensure that your application is properly signed and can be trusted by macOS. Here's a general outline of the process:

1. **Obtain an Apple Developer Account**: You need to have a valid Apple Developer account to create the necessary certificates and provisioning profiles.

2. **Generate a Developer Certificate**: In your Apple Developer account, generate a Developer Certificate for code signing. This certificate will be used to sign your application.

3. **Create an App Identifier**: Create an App Identifier for your Java application in your Apple Developer account. This identifier uniquely identifies your application.

4. **Generate a Provisioning Profile**: Create a provisioning profile that links your Developer Certificate and App Identifier. This profile allows your application to run on specific devices and be distributed.

5. **Configure Your Java Application**:
   - Ensure that your Java application's JAR files and resources are correctly organized within the DMG package.
   - Create a macOS `.app` bundle for your Java application. This involves structuring your application's files in a specific format that macOS expects.
   - Include an `Info.plist` file within the `.app` bundle. This file contains metadata about your application.

6. **Code Signing**:
   - Use the `codesign` command-line tool to sign the individual files within your `.app` bundle. This includes signing the JAR files and other executable components.
   - Use the Developer Certificate you generated earlier to sign the application.

7. **Packaging the DMG**:
   - Create a Disk Image (DMG) that will contain your signed application.
   - Place your signed `.app` bundle inside the DMG.
   - You can also include other resources such as icons, a background image, and a shortcut to the Applications folder.

8. **Testing**:
   - Before distributing your DMG, test the signed application on various macOS systems to ensure that it runs correctly and the code signing is working as expected.

9. **Distribution**:
   - Distribute the signed DMG package through your preferred distribution method (website, download link, etc.).
   - Users should be able to download and install the application by dragging it into the Applications folder.

**Comment by talk2emmzy** [comment]
> Hello @jainchirag21064 - could you try the following on your MacOS system? It may solve your issue, it worked for me:
> 
> * download the .dmg (I used the latest ZAP_2.9.0.dmg)
> * open it and drag 'OWASP ZAP' to the application directory as normal, wait for copying to finish
> * run ZAP from launchpad or Finder -> Applications
> * most likely get fatal pop-up _“OWASP ZAP” cannot be opened because the developer cannot be verified, macOS cannot verify that this app is free from malware._
> * close this message popup
> * from the MacOS 'Apple', top left of display, go to "System Preferences" > "Security & Privacy"
> * at the bottom of the dialog, see message saying that "OWASP ZAP" was blocked. Next to it, click "Open anyway"
> * the initial error message will pop up again, but this time have the option to click "Open" to run ZAP despite the warning
> * click 'Open' and ZAP now runs OK
> * this only has to be done once, after this ZAP can be run as normal

Thanks so much, this was helpful and worked for me on my MacBook Pro 💯

**Comment by hchasens** [comment]
While this works as a temporary solution, I think it'd be appropriate for Zap to follow best practices and sign their software such that it can be used without security alerts on Mac OSX.

**Comment by thc202** [comment]
That's the reason why the issue is still open.

**Comment by wavded** [comment]
Some MacOS EPM software like CyberArk prevents the running of this application altogether unless it is signed. Would be great if it would!

**Comment by haejinjo** [comment]
I just went to system settings and clicked open anyway.

**Comment by wavded** [comment]
homebrew is now saying:

Warning: zap has been deprecated because it does not pass the macOS Gatekeeper check! It will be disabled on 2026-09-01.
==> Fetching downloads for: zap
✔︎ Cask zap (2.17.0)

**Comment by benjuade** [comment]
Now it has been disabled: https://formulae.brew.sh/cask/zap
The solution may be to publish a dedicated Tap as Flameshot has done: https://github.com/flameshot-org/homebrew-flameshot
